example/src/App.tsx

```diff
@@ -1,27 +1,183 @@
 import { Netwatch } from 'react-native-netwatch';
 import { connect, Provider } from 'react-redux';
 import store from './redux/store';
-import { actionRequest } from './redux/actions/appActions';
 import { Dispatch } from 'redux';
+import { Provider as PaperProvider } from 'react-native-paper';
+import { Text, TouchableHighlight, StyleSheet, View } from 'react-native';
 
-const App = () => {
-  useEffect(() => {});
+// FIXME: RCTBridge required dispatch_sync to load RCTDevLoadingView. This may lead to deadlocks (iOS)
+
+const App = (props) => {
+  const [netwatchVisible, setNetwatchVisible] = useState(false);
+  const [netwatchEnabled, setNetwatchEnabled] = useState(true);
 
   return (
     <Provider store={store}>
-      <ConnectedComponent />
+      <PaperProvider>
+        <ConnectedComponent enabled={netwatchEnabled} visible={netwatchVisible} onPressClose={setNetwatchVisible} />
+        <View style={styles.container}>
+          <TouchableHighlight
+            style={styles.openButton}
+            onPress={() => {
+              setNetwatchVisible(true);
+            }}
+            testID="buttonDisplayNetwatch"
+          >
+            <Text style={styles.textStyle}>Display Netwatch</Text>
+          </TouchableHighlight>
+          <TouchableHighlight
+            style={styles.enableButton}
+            onPress={() => {
+              setNetwatchEnabled(!netwatchEnabled);
+            }}
+            testID="buttonDisabledNetwatch"
+          >
+            <Text style={styles.textStyle}>{netwatchEnabled ? 'Disabled Netwatch' : 'Enabled Netwatch'}</Text>
+          </TouchableHighlight>
+          <ConnectedButton />
+          
+        </View>
+      </PaperProvider>
     </Provider>
   );
 };
 
+const Button = (props) => {
+  return <TouchableHighlight
+    style={styles.enableButton}
+    onPress={() => {
+      props.customAction({ type: 'todos/todoAdded', payload: 'Learn about actions' });
+    }}
+    testID="buttonDisabledNetwatch"
+  >
+    <Text style={styles.textStyle}>Dispatch Action</Text>
+  </TouchableHighlight>
+
+}
+
 export function mapDispatchToProps(dispatch: Dispatch, props: any): any {
   return {
     ...props,
-    customAction: () => dispatch(actionRequest()),
+    customAction: (action) => dispatch(action),
   };
 }
 
+const ConnectedButton = connect(null, mapDispatchToProps)(Button);
 const ConnectedComponent = connect(null, mapDispatchToProps)(Netwatch);
 
 export default App;
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    paddingHorizontal: 16,
+  },
+  openButton: {
+    backgroundColor: '#F194FF',
+    borderRadius: 20,
+    padding: 10,
+    elevation: 2,
+    marginBottom: 16,
+  },
+  enableButton: {
+    backgroundColor: '#349EEB',
+    borderRadius: 20,
+    padding: 10,
+    elevation: 2,
+    marginBottom: 16,
+  },
+  textStyle: {
+    color: 'white',
+    fontWeight: 'bold',
+    textAlign: 'center',
+  },
+});
+
+// For testing only
```

Comment:
we can move this part outside of App.tsx

Reply:
Done